File: YYWButtonCell/buttoncell/ViewController.m
//
//  ViewController.m
//  buttoncell
//
//  Created by 赵睿 on 12/4/17.
//  Copyright © 2017年 赵睿. All rights reserved.
//

#import "ViewController.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // Do any additional setup after loading the view.
}


- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}

- (IBAction)toggleMainButton:(NSButton *)sender {
    if (sender.state == NSOnState) {
        self.mainButton.enabled = YES;
    } else if (sender.state == NSOffState) {
        self.mainButton.enabled = NO;
    } else {
        //mixed state
    }
}

@end

File: Lesson54/ViewController.swift
//
//  ViewController.swift
//  Lesson54
//
//  Created by 赵睿 on 21/01/2017.
//  Copyright © 2017 com.homebrew.zhaorui. All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    @IBOutlet weak var blue_button: NSButton!
    @IBOutlet weak var yellow_button: NSButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        blue_button.title = "Button"
        
        //set below to true, yellow cell won't draw the frame
        yellow_button.showsBorderOnlyWhileMouseInside = true;
    }
    

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }


}

